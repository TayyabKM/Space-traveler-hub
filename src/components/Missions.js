import React from 'react';

export default function Missions() {
  return (
    <div>
      <h1> HELLO MISSIONS</h1>

    </div>
  );
}
