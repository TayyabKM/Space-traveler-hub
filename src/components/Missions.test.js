import React from 'react';
import { useSelector } from 'react-redux';
import { render } from '@testing-library/react';
import Missions from './Missions';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

describe('Missions component', () => {
  it('matches snapshot', () => {
    useSelector.mockImplementation((selector) => selector({
      missions: [
        {
          mission_id: '1',
          mission_name: 'Mars Mission',
          description: 'A mission to Mars',
          reserved: false,
        },
      ],
    }));

    const { asFragment } = render(<Missions />);
    expect(asFragment()).toMatchSnapshot();
  });
});
