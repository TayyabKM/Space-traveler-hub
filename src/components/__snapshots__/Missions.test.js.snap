// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mission-container"
  >
    <div
      class="titles"
    >
      <h2
        class="border-container"
      >
        Mission
      </h2>
      <h2
        class="border-container"
      >
        Description
      </h2>
      <h2
        class="border-container"
      >
        Status
      </h2>
      <h2
        class="border-container"
      >
         
      </h2>
    </div>
    <ul
      class="mission-list"
    >
      <li
        class="mission-item"
      >
        <div
          class="border-container"
        >
          <h3
            class="mission-name"
          >
            Mars Mission
          </h3>
        </div>
        <div
          class="border-container"
        >
          <p
            class="mission-description"
          >
            A mission to Mars
          </p>
        </div>
        <div
          class="border-container"
        >
          <div
            class="mission-badge-container"
          >
            <div
              class="not-a-member-badge"
            >
              NOT A MEMBER
            </div>
          </div>
        </div>
        <div
          class="border-container"
        >
          <div
            class="mission-action-container"
          >
            <button
              class="join-mission-button"
              type="button"
            >
              Join Mission
            </button>
          </div>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
